lib/pymedphys/tests/dicom/test_connect.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (C) 2020 University of New South Wales & Ingham Institute
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=redefined-outer-name
+
+import shutil
+from unittest.mock import Mock
+
+import pytest
+
+import pydicom
+from pynetdicom import AE, VerificationPresentationContexts
+from pynetdicom.sop_class import RTPlanStorage  # pylint: disable=no-name-in-module
+
+from pymedphys._dicom.connect.listen import DicomListener
+from pymedphys._dicom.create import dicom_dataset_from_dict
+
+# TODO How to determine an appropriate port for testing?
+TEST_PORT = 9988
+
+METHOD_MOCK = Mock()
+
+
+@pytest.fixture()
+def listener():
+
+    dicom_listener = DicomListener(
+        port=TEST_PORT, on_released_callback=METHOD_MOCK.method
+    )
+    dicom_listener.start()
+
+    yield dicom_listener
+
+    dicom_listener.stop()
+
+
+@pytest.mark.pydicom
+def test_dicom_listener_echo(listener):
+    """Test to ensure that running dicom listener responds to C-ECHO
+    """
+
+    # Send C-ECHO
+    ae = AE()
+    ae.requested_contexts = VerificationPresentationContexts
+
+    assoc = ae.associate(listener.host, listener.port, ae_title=listener.ae_title)
+
+    result = None
+    if assoc.is_established:
+        status = assoc.send_c_echo()
+
+        if status:
+            result = status.Status
+
+        assoc.release()
+
+    # Check we got a valid result
+    assert result == 0
+
+
+@pytest.mark.pydicom
+def test_dicom_listener_send(listener):
+    """Test to ensure that running dicom listener recieves a stores a Dicom file
+    """
+
+    METHOD_MOCK.reset_mock()
+
+    # Create a test Dicom object
+    test_uid = pydicom.uid.generate_uid()
+    test_dataset = dicom_dataset_from_dict(
+        {
+            "SOPClassUID": RTPlanStorage,
+            "SOPInstanceUID": test_uid,
+            "SeriesInstanceUID": test_uid,
```

Comment:
it's better to not have two UIDs being the same, even if one is for the Series and one for the Instance (unless of course they are intentionally the same Study or Series UID indicating common heritage).